govcd/media.go

```diff
@@ -354,3 +413,220 @@ func (mediaItem *MediaItem) Refresh() error {
 
 	return err
 }
+
+// Refresh refreshes the media information by href
+func (media *Media) Refresh() error {
+
+	if media.Media == nil {
+		return fmt.Errorf("cannot refresh, Object is empty")
+	}
+
+	url := media.Media.HREF
+
+	// Empty struct before a new unmarshal, otherwise we end up with duplicate
+	// elements in slices.
+	media.Media = &types.Media{}
+
+	_, err := media.client.ExecuteRequest(url, http.MethodGet,
+		"", "error retrieving media: %s", nil, media.Media)
+
+	return err
+}
+
+// GetMediaByHref finds a Media by HREF
+// On success, returns a pointer to the Media structure and a nil error
+// On failure, returns a nil pointer and an error
+func (cat *Catalog) GetMediaByHref(mediaHref string) (*Media, error) {
+
+	media := NewMedia(cat.client)
+
+	_, err := cat.client.ExecuteRequest(mediaHref, http.MethodGet,
```

Comment:
If we get an explicit response here (instead of `_`), we can check whether the status code is 404, and in that case return `ErrEntityNotFound`. This way, `GetMediaById` doesn't need to be changed (apart from returning the correct errors on parsing failures) and the entity test can be used.

Reply:
We get `explicit` Response status 403 Forbidden - which is quite common in API in such cases. If we agree - check would be simple.

Reply:
I was expecting to get a 404.
I'd like to see an explicit test for this.

Reply:
Usually it's security feature to return 403. 404 already tells information.